### PR TITLE
Update Mongoose to latest version

### PR DESCRIPTION
**Pull request by dmskr**
Bring also latest changes from stable branch

### CODE REVIEW

File: test/fortune/should-upsert.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ module.exports = function(options){
 
     function createDuplicate(){
       return new RSVP.Promise(function(resolve){
-        request(baseUrl).post('/people')
+        request("http://127.0.0.1:8891").post('/people')
```
Comment (EugeneKostrikov) [inline test/fortune/should-upsert.js]: Hardco~~re~~de

File: test/fortune/routing.js [hunk 2]
```diff
@@ -71,6 +71,7 @@ module.exports = function(options){
           .end(function(err, res){
             should.not.exist(err);
             var body = JSON.parse(res.text);
+            console.log(body);
```
Comment (EugeneKostrikov) [inline test/fortune/routing.js]: Still needed here?

File: test/fortune/fields_and_filters.js [hunk 3]
```diff
@@ -159,7 +159,7 @@ module.exports = function(options){
             done();
           });
       });
-      it("should return correct error if date is not recognized", function(done) {
+      it.skip("should return correct error if date is not recognized", function(done) {
```
Comment (EugeneKostrikov) [inline test/fortune/fields_and_filters.js]: `.skip`

File: test/fortune/fields_and_filters.js [hunk 4]
```diff
@@ -196,7 +196,7 @@ module.exports = function(options){
             done();
           });
       });
-      it("should return correct error if date is not recognized inside of complex filter with $gte, $lte", function(done) {
+      it.skip("should return correct error if date is not recognized inside of complex filter with $gte, $lte", function(done) {
```
Comment (EugeneKostrikov) [inline test/fortune/fields_and_filters.js]: `.skip`

File: lib/adapters/mongodb.js [hunk 5]
```diff
@@ -836,7 +837,7 @@ adapter._updateOneToMany = function(model, relatedModel, resource, reference, fi
       match[relatedModel.pk || "_id"] = resource[reference.path];
       if (resource._tenantId) match._tenantId = resource._tenantId;
 
-      return wrapAsyncCall(relatedModel, relatedModel.findOneAndUpdate, match, associate);
+      return wrapAsyncCall(relatedModel, relatedModel.findOneAndUpdate, match, associate, { new: true });
```
Comment (EugeneKostrikov) [inline lib/adapters/mongodb.js]: What does `{new: true}` change here? Presume it's something to do with [this issue](https://github.com/Automattic/mongoose/issues/2262) but don't quite follow why would it matter for fortune

File: package.json [hunk 6]
```diff
@@ -18,27 +18,28 @@
     "i": "~0.3.2",
     "lodash": "3.10.1",
     "mkdirp": "0.5.1",
+    "mocha": "^3.0.0",
     "moment": "2.10.6",
-    "mongoose": "~3.8.21",
+    "mongoose": "^4.5.8",
```
Comment (EugeneKostrikov) [inline package.json]: Lets better freeze the versions on exact values (as in `"4.5.8"`). At least for everything we explicitly upgrade to and test

File: test/fortune/fields_and_filters.js (hunk 3 above)
Comment (dmskr) [inline test/fortune/fields_and_filters.js]: Fixed

File: test/fortune/fields_and_filters.js (hunk 4 above)
Comment (dmskr) [inline test/fortune/fields_and_filters.js]: Fixed

File: test/fortune/routing.js (hunk 2 above)
Comment (dmskr) [inline test/fortune/routing.js]: Removed

File: test/fortune/should-upsert.js (hunk 1 above)
Comment (dmskr) [inline test/fortune/should-upsert.js]: Removed

File: lib/adapters/mongodb.js (hunk 5 above)
Comment (dmskr) [inline lib/adapters/mongodb.js]: Latest version of mongoose follows mongo native driver practice to return nothing after update (or return the data you just provided in the update). The param  `{ new: true }` forces it to re-read whatever data in database and return the record, as most of the fortune expects such a behaviour.

File: package.json (hunk 6 above)
Comment (dmskr) [inline package.json]: Done

**Pull request merged**